local-cluster/tests/local_cluster.rs

```diff
@@ -4480,6 +4480,226 @@ fn test_slot_hash_expiry() {
+        leader_schedule: Arc::new(leader_schedule),
+    });
+    let vs = validator_keys
+        .iter()
+        .map(|kp| kp.pubkey())
+        .collect::<Vec<_>>();
+    let (a_pubkey, b_pubkey, c_pubkey) = (vs[0], vs[1], vs[2]);
+
+    // We want B to not vote (we are trying to simulate its votes not landing until it gets to the
+    // minority fork)
+    let mut validator_configs =
+        make_identical_validator_configs(&default_config, node_stakes.len());
+    validator_configs[1].voting_disabled = true;
+
+    let mut config = ClusterConfig {
+        cluster_lamports: DEFAULT_CLUSTER_LAMPORTS + node_stakes.iter().sum::<u64>(),
+        node_stakes,
+        validator_configs,
+        validator_keys: Some(
+            validator_keys
+                .iter()
+                .map(|s| (s.clone(), true))
+                .collect::<Vec<_>>(),
+        ),
+        node_vote_keys: Some(node_vote_keys),
+        slots_per_epoch,
+        stakers_slot_offset: slots_per_epoch,
+        skip_warmup_slots: true,
+        ..ClusterConfig::default()
+    };
+    let mut cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
+
+    let mut common_ancestor_slot = 8;
+
+    let a_ledger_path = cluster.ledger_path(&a_pubkey);
+    let b_ledger_path = cluster.ledger_path(&b_pubkey);
+
+    // Immediately kill B and C (we just needed it for the initial stake distribution)
+    info!("Killing B and C");
+    let mut b_info = cluster.exit_node(&b_pubkey);
+    cluster.exit_node(&c_pubkey);
+
+    // Let A run for a while until we get to the common ancestor
+    info!("Letting A run until common_ancestor_slot");
+    let now = Instant::now();
+    loop {
+        let last_vote = wait_for_last_vote_in_tower_to_land_in_ledger(&a_ledger_path, &a_pubkey);
+        if last_vote >= common_ancestor_slot {
+            common_ancestor_slot = last_vote;
+            info!("common_ancestor_slot: {}", common_ancestor_slot);
+            break;
+        }
+        assert!(
+            now.elapsed() < Duration::from_secs(10),
+            "validator not creating blocks fast enough"
+        );
+        sleep(Duration::from_millis(100));
+    }
+
+    // Copy A's ledger to B but truncate any blocks past the common ancestor.
+    {
+        info!("Copying A's ledger to B");
+        std::fs::remove_dir_all(&b_info.info.ledger_path).unwrap();
+        let mut opt = fs_extra::dir::CopyOptions::new();
+        opt.copy_inside = true;
+        fs_extra::dir::copy(&a_ledger_path, &b_ledger_path, &opt).unwrap();
+
+        // remove A's tower in B's new copied ledger
+        info!("Removing A's tower in B's ledger dir");
+        remove_tower(&b_ledger_path, &a_pubkey);
+
+        // Get rid of any slots past common_ancestor_slot
+        info!("Removing extra slots from B's blockstore");
+        let blockstore = open_blockstore(&b_ledger_path);
+        purge_slots_with_count(&blockstore, common_ancestor_slot + 1, 100);
```

Comment:
Think this can be simplified by only copying blocks from `A`

```suggestion
        info!("Copying A's ledger to B");
        let blockstore_a= open_blockstore(&a_ledger_path);
        let blockstore_b = open_blockstore(&b_ledger_path);
        copy_blocks(common_ancestor_slot, &blockstore_a, &blockstore_b);
```